jbmc/unit/java_bytecode/java_bytecode_convert_method/convert_method.cpp

```diff
@@ -286,3 +288,84 @@ SCENARIO(
     }
   }
 }
+
+/// Allow access to private methods so that they can be unit tested
+class java_bytecode_convert_method_unit_testt
+{
+public:
+  static exprt
+  convert_aload(const irep_idt &statement, const exprt::operandst &op)
+  {
+    return java_bytecode_convert_methodt::convert_aload(statement, op);
+  }
+};
+
+TEST_CASE(
+  "baload byte array",
+  "[core][java_bytecode][java_bytecode_convert_method][convert_aload]")
+{
+  const typet byte_array_type = java_array_type('b');
+  const symbol_exprt byte_array{"byte_array", byte_array_type};
+  const exprt offset = from_integer(1, java_int_type());
+  const exprt result = java_bytecode_convert_method_unit_testt::convert_aload(
+    "baload", {byte_array, offset});
+
+  // Result is of the form `(int)(*(byte_array->data + offset))`
+  auto query = make_query(result)
+                 .as<typecast_exprt>()[0]
```

Comment:
:question: Why do we cast to int?

Reply:
This is documented in `java_bytecode_promotion`, I've added a reference to that here.